### PR TITLE
Run instrumentation tests on Firebase Test Lab (Dev)

### PR DESCRIPTION
**Pull request by mtwalli**
Run Instrumentation tests on Firebase TL 🔥 

Result:
🟢 Tests + Enjoy 👨🏼‍💻

### CODE REVIEW

**Review by AlexanderAlferov — DISMISSED** [review]
Great! 🚀

File: .circleci/config.yml [hunk 1]
```diff
@@ -437,35 +450,54 @@ jobs:
...
-          desc: Run instrumentation tests on device
-          cmd: >
-            :Corona-Warn-App:connectedDeviceDebugAndroidTest
-            --info --continue
-            -Pandroid.testInstrumentationRunnerArguments.notAnnotation=testhelpers.Screenshot
-            -Pandroid.testInstrumentationRunnerArguments.clearPackageData=true
-      - kill-all-emulators
+      - restore-android-build-cache
+      - run:
+          name: "Build app APK "
```
Comment (d4rken) [inline .circleci/config.yml:464]: Extra space, only remove it if you change the file for any other reason.

File: .circleci/config.yml [hunk 2]
```diff
@@ -437,35 +450,54 @@ jobs:
...
+          command: ./gradlew -PdisablePreDex :Corona-Warn-App:assembleDeviceForTestersDebug
+      - run:
```
Comment (d4rken) [inline .circleci/config.yml:466]: Could we use our `run-gradle-cmd` which automatically adds a few optimization flags, or did you try that and there were downsides?

File: .circleci/config.yml [hunk 3]
```diff
@@ -437,35 +450,54 @@ jobs:
...
+              --no-record-video
+          no_output_timeout: 30m
+      - run:
+          name: Create directory to store test results
+          command: mkdir ~/project/firebase
+      - run:
+          name: Install gsutil dependency and copy test results data
+          command: |
+            sudo pip install -U crcmod
+            sudo gsutil -m cp -R -U "`sudo gsutil ls gs://${GOOGLE_PROJECT_ID}-circleci-android/${BUCKETDIR} | tail -1`*"  ~/project/firebase | true
```
Comment (d4rken) [inline .circleci/config.yml:496]: I think we could make the path simpler.
The generated path in the zip is currently `/home/circleci/project/firebase/flame-29-de_DE-portrait/`

We could try

```yaml
          sudo gsutil -m cp -R -U "`sudo gsutil ls gs://${GOOGLE_PROJECT_ID}-circleci-android/${BUCKETDIR} | tail -1`*" firebase-results | true

(...)
      - store_test_results:
          path: ./firebase-results/flame-29-de_DE-portrait
      - compress-path:
          input: ./firebase-results/flame-29-de_DE-portrait/*.xml
(...)
```

What do you think?

**Review by d4rken — COMMENTED** [review]
Looks good, few comments.

File: .circleci/config.yml [hunk 4]
```diff
@@ -437,35 +450,54 @@ jobs:
             --form "file=@${fileName}" \
 
   instrumentation_tests_device:
-    macos:
-      xcode: "12.3.0"
-    resource_class: large
+    resource_class: xlarge
```
Comment (d4rken) [inline .circleci/config.yml:453]: Instrumentation tests are now faster than the overall longest job-chain (unit tests + sonar), so we could save some resources by preventing firebase from spinning up if trivial checks fails, i.e.

```yml
      - instrumentation_tests_device:
          requires:
            - ktlint_device_release_check
            - detekt
```

Thoughts?

Comment (mtwalli) [reply .circleci/config.yml:453]: Good point

Comment (mtwalli) [reply .circleci/config.yml:453]: Good point

File: .circleci/config.yml (hunk 2 above)
Comment (mtwalli) [reply .circleci/config.yml]: will try it

**Comment by mtwalli** [comment]
Comments are done, Please have another look

**Review by d4rken — APPROVED** [review]
lgtm, thank you for optimizing this

**Review by ralfgehrer — APPROVED** [review]

**Pull request merged**